akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/voice/VoiceApp.scala

```diff
@@ -136,6 +136,19 @@ object VoiceApp {
               // Update the user status to indicate they are still in the voice conference.
               VoiceUsers.setLastStatusUpdate(liveMeeting.voiceUsers, vu)
             }
+
+            // Purge voice users that don't have a matching user record
+            // Avoid this if the meeting is a breakout room since might be real
+            // voice users participating
+            if (ejectRogueVoiceUsers && !liveMeeting.props.meetingProp.isBreakout) {

```

Comment:
I think we need to add a check here if the user is not a dial-in user then look for the user in the users list.